data/transactions/logic/eval.go

```diff
@@ -1699,6 +1699,36 @@ func opDig(cx *evalContext) {
 	cx.stack = append(cx.stack, sv)
 }
 
+func opCover(cx *evalContext) {
+	depth := int(cx.program[cx.pc+1])
+	idx := len(cx.stack) - 1 - depth
+	// Need to check stack size explicitly here because checkArgs() doesn't understand cover
+	// so we can't expect our stack to be prechecked.
+	if idx < 0 {
+		cx.err = fmt.Errorf("cover %d with stack size = %d", depth, len(cx.stack))
+		return
+	}
+	topIdx := len(cx.stack) - 1
```

Comment:
This may read better if you move `topIdx` above `idx` like so:
```go
topIdx := len(cx.stack - 1)
idx := topIdx - depth
```

Reply:
good point